Drawsana/Data model/Drawing.swift

```diff
@@ -83,20 +95,25 @@ public class Drawing: Codable {
         // we can report exactly what kind of thing can't be parsed
         let typeContainer = try shapeIter.nestedContainer(keyedBy: ShapeTypeCodingKey.self)
         let type = try typeContainer.decode(String.self, forKey: .type)
-        fatalError("Can't decode shape of type \(type)")
+
+        // If in debug mode, this is an error. Otherwise, just ignore it and
+        // move on.
+        if Drawing.debugSerialization {
+          throw DrawsanaDecodingError.unknownShapeTypeError(type)
+        }
       }
     }
   }
 
-  private func tryDecodingAllShapes(_ container: inout UnkeyedDecodingContainer) -> [Shape] {
+  private func tryDecodingAllShapes(_ container: inout UnkeyedDecodingContainer) throws -> [Shape] {
```

Comment:
[nit] We might want to remove `try` from this method name as well as `tryDecoding()` now that they throw errors. It looks weird that the calls are e.g. `try tryDecodingAllShapes(...)`.